zplugin-ros1/src/ros_to_zenoh_bridge/zenoh_client.rs

```diff
@@ -0,0 +1,148 @@
+    // PUBLIC
+    pub fn new(session: Arc<Session>) -> ZenohClient {
+        ZenohClient { session }
+    }
+
+    pub async fn subscribe<C>(
+        &self,
+        key_expr: &str,
+        callback: C,
+    ) -> ZResult<zenoh::subscriber::Subscriber<'static, ()>>
+    where
+        C: Fn(Sample) + Send + Sync + 'static,
+    {
+        debug!("Creating Subscriber on {}", key_expr);
+
+        match self.make_keyexpr(key_expr).await {
+            Ok(opt_keyexpr) => {
+                self.session
+                    .declare_subscriber(opt_keyexpr)
+                    .callback(callback)
+                    .allowed_origin(Locality::Remote)
+                    .res_async()
+                    .await
+            }
+            Err(e) => Err(e),
+        }
+    }
+
+    pub async fn publish(&self, key_expr: &str) -> ZResult<zenoh::publication::Publisher<'static>> {
+        debug!("Creating Publisher on {}", key_expr);
+
+        match self.make_keyexpr(key_expr).await {
+            Ok(opt_keyexpr) => {
+                self.session
+                    .declare_publisher(opt_keyexpr)
+                    .allowed_destination(Locality::Remote)
+                    .congestion_control(CongestionControl::Block)
+                    .res_async()
+                    .await
+            }
+            Err(e) => Err(e),
+        }
+    }
+
+    pub async fn make_queryable<Callback>(
+        &self,
+        key_expr: &str,
+        callback: Callback,
+    ) -> ZResult<zenoh::queryable::Queryable<'static, ()>>
+    where
+        Callback: Fn(zenoh::queryable::Query) + Send + Sync + 'static,
+    {
+        info!("Creating Queryable on {}", key_expr);
+
+        match self.make_keyexpr(key_expr).await {
+            Ok(opt_keyexpr) => {
+                self.session
+                    .declare_queryable(opt_keyexpr)
+                    .allowed_origin(Locality::Remote)
+                    .callback(callback)
+                    .res_async()
+                    .await
+            }
+            Err(e) => Err(e),
+        }
+    }
+
+    pub async fn make_query<Callback>(
+        &self,
+        key_expr: &str,
+        callback: Callback,
+        data: Vec<u8>,
+    ) -> ZResult<()>
+    where
+        Callback: Fn(zenoh::query::Reply) + Send + Sync + 'static,
+    {
+        debug!("Creating Query on {}", key_expr);
+
+        match self.make_keyexpr(key_expr).await {
+            Ok(opt_keyexpr) => {
+                self.session
+                    .get(opt_keyexpr)
+                    .with_value(data)
+                    .callback(callback)
+                    .res_async()
+                    .await
+            }
+            Err(e) => Err(e),
+        }
+    }
+
+    pub async fn make_query_sync(
+        &self,
+        key_expr: &str,
+        data: Vec<u8>,
+    ) -> ZResult<flume::Receiver<zenoh::query::Reply>> {
+        debug!("Creating Query on {}", key_expr);
+
+        match self.make_keyexpr(key_expr).await {
```

Comment:
`make_keyexpr` is not necessary since the `get` will already:
1) verify the correct format of the `keyexpr`;
2) use an exisiting keyexpr being declared by an existing queryable.

Moreover, keep in mind that any Zenoh API that starts with `declare` has an impact on the infrastructure and network. So, it's not wise to declare things at every query.